src/libraries/JANA/Services/JParameterManager.h

```diff
@@ -267,6 +270,26 @@ JParameter* JParameterManager::SetDefaultParameter(std::string name, T& val, std
     return param;
 }
 
+/// @brief Retrieve a configuration parameter, if necessary creating it with the provided default value.
+///
+/// @param [in] name            The parameter name. Must not contain spaces.
+/// @param [in,out] default_val Value to set the desired default value to (returned if no value yet set for parameter).
+/// @param [in] description     Optional description, e.g. units, set or range of valid values, etc.
+/// @return                     Current value of parameter
+///
+/// @details This is a convenience method that wraps SetDefaultParameter. The difference
+/// is that this has the default passed by value (not by reference) and the value of the parameter is returned
+/// by value. This allows a slightly different form for declaring configuration parameters with a default value.
+/// e.g.
+///     auto thresh = jpp->RegisterParameter("SystemA:threshold", 1.3, "threshold in MeV");
+///
+template <typename T>
+inline T JParameterManager::RegisterParameter(std::string name, const T default_val, std::string description){
+    T val = default_val;
+    SetDefaultParameter(name.c_str(), val, description);
```

Comment:
How can this work? Aren't you saving address to a variable on stack?

Reply:
(I was assuming that the parameters can be updated at runtime, not just before plugin load)

Reply:
```suggestion
    SetDefaultParameter(name, val, description);
```

Reply:
No. The JParameter object maintains its own members that hold the value and copy it into user-provided variables (passed by reference) upon request. It does not do us much good to keep references to variables in factories since we could never change them, The factories are being run asynchronously so changing them on-the-fly could result in non-reproducable results.